internal/controllers/machinedeployment/machinedeployment_controller.go

```diff
@@ -91,14 +90,8 @@ func (r *Reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, opt
 		WithEventFilter(predicates.ResourceNotPausedAndHasFilterLabel(ctrl.LoggerFrom(ctx), r.WatchFilterValue)).
 		Watches(
 			&clusterv1.Cluster{},
+			// TODO: should this wait for Cluster.Status.InfrastructureReady similar to Infra Machine resources?
 			handler.EnqueueRequestsFromMapFunc(clusterToMachineDeployments),
-			builder.WithPredicates(
-				// TODO: should this wait for Cluster.Status.InfrastructureReady similar to Infra Machine resources?
-				predicates.All(ctrl.LoggerFrom(ctx),
-					predicates.ClusterUnpaused(ctrl.LoggerFrom(ctx)),
```

Comment:
I think ClusterUnpaused checks for Cluster.spec.paused while the other one only checks the annotation

Reply:
I was somehow under the impression that it checks both spec.paused (if available) and paused annotations